sympy/functions/elementary/trigonometric.py

```diff
@@ -222,13 +224,30 @@ def fdiff(self, argindex=1):
 
     @classmethod
     def eval(cls, arg):
+        from sympy.calculus import Limits
         if arg.is_Number:
             if arg is S.NaN:
                 return S.NaN
             elif arg is S.Zero:
                 return S.Zero
             elif arg is S.Infinity or arg is S.NegativeInfinity:
-                return
+                return Limits(-1, 1)
+
+        if isinstance(arg, Limits):
+            start, end = arg.start, arg.end
```

Comment:
``start`` and ``end`` are attributes of ``Interval``. As ``Limit`` is now as a subclass of ``Interval`` there might be some confusion. Does this mean that ``min`` and ``max`` should be renamed (again) to be on the safe side?